vassal-app/src/main/java/VASSAL/configure/ConfigureTree.java

```diff
@@ -1479,6 +1716,74 @@ private int getNumMatches(String searchString) {
       return (int) searchNodes.stream().filter(node -> checkNode(node, searchString)).count();
     }
 
+
+    /**
+     * @param st - Search target (usually Decorator or AbstractConfigurable)
+     * @param searchString - our search string
+     * @return true if the node matches our searchString based on search configuration ("match" checkboxes)
+     */
+    private boolean checkSearchTarget(SearchTarget st, String searchString) {
+      if (searchParameters.isMatchExpressions()) {
+        List<String> exps = st.getExpressionList();
+        if (exps != null) {
+          for (String s : exps) {
+            if (!StringUtils.isEmpty(s) && checkString(s, searchString)) {
+              return true;
+            }
+          }
+        }
+      }
+
```

Comment:
It just occurred to me that this is a linear search over all the crap every time... Is this insanely slow?

Reply:
It is essentially instantaneous! For I guess the same reason that text editors search fast :)

Reply:
You mean that there's just not very much input?

Reply:
It's essentially like text-searching forward through the XML file itself -- it's just not that big.

Reply:
I guess I'm just used to dealing with TB of data for searches...